### DISCUSSION TITLE
Is there a way to see if value returned is retrned from Cache or Db

### DISCUSSION THREAD

**Discussion by Amorganskate** [ZiggyCreatures/FusionCache#274] [General]
Is there a way to see if value returned is retrned from Cache or Db? 

Might end up running into a hard requirement to return a response header letting client know if cache was hit or not

**Comment by jodydonetti** [comment]
Hi @Amorganskate no, there's not, at least not _built in_.

This is because the return type of the `GetOrSet` methods is the `TValue` itself, so there's no "space" to put an hypothetical "fromCache" bool flag or similar.

What you may do is use an approach like this:

```csharp
var fromCache = true;
var person = cache.GetOrSet<Person>(
  "person:1",
  (ctx, ct) => {
    var res = GetPersonFromDb(1);
    fromCache = false; // THIS HERE
    return res;
  },
  ...
);

if (fromCache) {
  // HERE YOU'LL KNOW THAT PERSON CAME FROM THE CACHE
  // ...
}
```

Hope this helps, let me know!

**Reply by Amorganskate** [reply]
We have some clients on production this week officially with the new fusion cache library, still resolving some custom json serialization methods but other then that this is had some crazy results for us. Went from about 230ms avg down to 34ms

**Reply by jodydonetti** [reply]
Can I say I'm really happy about it?
Yes, yes I can 😬
